src/main/java/athena/TaskList.java

```diff
@@ -65,18 +75,27 @@ public Task markTaskAsDone(int taskNumber) {
     /**
      * Adds a task to the task list.
      *
-     * @param name Name of task
-     * @param startTime Start time of task
-     * @param duration Duration of task
-     * @param deadline Deadline of task
+     * @param name       Name of task
+     * @param startTime  Start time of task
+     * @param duration   Duration of task
+     * @param deadline   Deadline of task
      * @param recurrence Recurrence of task
      * @param importance Importance of task
-     * @param notes Additional notes of task
+     * @param notes      Additional notes of task
      */
+    public void addTask(String name, String startTime, String duration,
+                        String deadline, String recurrence, String importance, String notes, int index) {
+
+        Task task = createTask(name, startTime, duration, deadline, recurrence, importance, notes, index);
+        tasks.add(task);
+    }
+
     public void addTask(String name, String startTime, String duration,
                         String deadline, String recurrence, String importance, String notes) {
+        this.maxIndex += 1;
+        int index = maxIndex;
 
-        Task task = createTask(name, startTime, duration, deadline, recurrence, importance, notes);
+        Task task = createTask(name, startTime, duration, deadline, recurrence, importance, notes, index);
```

Comment:
Repeated code. Can do `addTask(name, startTime, duration, deadline, recurrence, importance, notes, index)`.

Reply:
The overloaded method is to distinguish tasks that require a new index vs a preloaded task with its own assigned index